src/get-ipfs.ts

```diff
@@ -21,125 +26,144 @@ interface IpfsPkg extends ipfs {
-  }
-  if (permissions.indexOf('version') < 0) {
-    permissions.push('version')
-  }
+  if (permissions.indexOf('id') < 0) permissions.push('id')
+  if (permissions.indexOf('version') < 0) permissions.push('version')
   return permissions
 }
 
-const connectPeers = async (ipfs: ipfs, peers: string[] = []) => {
-  await Promise.all(
-    peers.map(async p => {
-      try {
-        await ipfs.swarm.connect(p)
-      } catch (err) {
-        console.log('Could not connect to peer: ', p)
-      }
-    })
-  )
-  return ipfs
-}
 
-export const ipfsIsWorking = async (ipfs?: ipfs) => {
-  if (!ipfs) {
-    return false
+
+// Default
+// =======
+
+const getIpfs = async (config: config = {}) => {
+  if (ipfsInstance) return ipfsInstance
+
+  // If not set up before, start flow.
+  config = { ...config, peers: config.peers || [] }
+  ipfsInstance = await loadWindowIpfs(config)
+
+  if (ipfsInstance) {
+    console.log('window.ipfs is available!')
+    const peers = (config.peers || []).concat(config.localPeers || [])
+    await connectPeers(ipfsInstance, peers)
+
+  } else {
+    console.log('window.ipfs is not available, loading js-ipfs...')
+    ipfsInstance = await loadJsIpfs(config)
+
   }
+
+  return ipfsInstance
+}
+
+
+export default getIpfs
+
+
+
+// 🛠
+
+
+export async function ipfsIsWorking(ipfs?: ipfs): Promise<boolean> {
+  if (!ipfs) return false
+
   try {
     const id = await ipfs.id()
-    if (id.id && id.agentVersion) {
-      return true
-    } else {
-      return false
-    }
+    return !!(id.id && id.agentVersion)
   } catch (err) {
     return false
   }
 }
 
-export const loadWindowIpfs = async (config: config): Promise<ipfs | null> => {
-  let ipfs = (window as IpfsWindow).ipfs
-  if (ipfs) {
-    if (ipfs.enable) {
-      const permissions = normalizePermissions(config.permissions)
-      ipfs = await ipfs.enable({
-        commands: permissions
-      })
-    }
-    const isWorking = await ipfsIsWorking(ipfs)
-    if (isWorking) {
-      return ipfs as ipfs
-    } else {
-      return null
-    }
-  }
-  return null
-}
 
-export const loadIpfsScript = (): Promise<void> => {
-  return new Promise((resolve, reject) => {
-    const script = document.createElement('script')
-    script.src = 'https://unpkg.com/ipfs@0.38.0/dist/index.min.js'
-    script.onload = async () => {
-      return resolve()
-    }
-    script.onerror = reject
-    document.body.appendChild(script)
-  })
+export async function loadJsIpfs(config: config): Promise<ipfs | null> {
+  const ipfsPkg = await (() => { switch (typeof config.jsIpfs) {
+    case "function":  return config.jsIpfs()
+    case "string":    return loadJsIpfsThroughScript(config.jsIpfs)
+    default:          return loadJsIpfsThroughScript(UNPKG_URL)
+  }})()
+
+  if (!ipfsPkg?.create) return null
+  const peers = (config.peers || []).concat(config.browserPeers || [])
+  const ipfs = await ipfsPkg.create({ config: { Addresses: { Swarm: peers, Bootstrap: peers }}})
```

Comment:
Not sure about this bootstrap thing, the js-ipfs docs say that peers added to the swarm also need to be in the bootstrap list. But, what I'm worried about is, that I guess js-ipfs has a default set of peers/nodes. Does this override the default set, or does it extend it?